fix: now cant delete course with student; Closes: #964

9921622 opened the pull request:
all i did was add models.PROTECT + quick test for that

tylerecouture commented on src/courses/tests/test_models.py:
```diff
@@ -323,6 +323,12 @@ def test_xp_per_day_ave(self, days_so_far):
+            quick test to see if cascade
+        """ 
+        self.assertRaises(Exception, self.course.delete)
+
```
This is in the wrong test class (CourseStudent instead of Course, and the docstring text appears to be cutoff.

9921622 replied:
Moved the test to CourseModelTest

tylerecouture reviewed (changes requested):
See my comment about test location.  Also, what's the user experience if a teacher tries to do this? Is it an error page? or what happens?  It needs to be informative to the user what's going one or why they can't delete the course.  Can you try it and see what happens?

9921622 commented:
Added:
- text when user tries to delete course with students
- prevent any forced post request on deleteview if it has any students
- added tests on course views

tylerecouture commented on src/courses/tests/test_models.py:
```diff
@@ -277,6 +277,17 @@ def test_default_object_created(self):
 
+    def test_model_protection(self):
+        """ 
+            quick test to see if model deleteion is protected when trying to delete progmatically
```
Two typos here (is spell check turned on in your IDE? Also, be more specific about what you are actually testing.  It's not clear to me from the docstring what you are testing specifically, which model? what's being deleted? what should happen?

9921622 replied:
fixed

9921622 replied:
didnt have a spell check will install one

tylerecouture commented on src/courses/views.py:
```diff
@@ -128,6 +128,24 @@ class CourseDelete(NonPublicOnlyViewMixin, DeleteView):
+        return context
+
+    def post(self, request, *args, **kwargs):
+        # make sure no data can be pased in POST if populated 
```
typos?  "pased"

9921622 replied:
fixed

tylerecouture commented on src/courses/templates/courses/course_confirm_delete.html:
```diff
@@ -4,14 +4,21 @@
+{% else %}
+  <p>Unable to delete '{{ course.title }}' as it still has students registered. Perhaps try to disable the <a href="{% url 'courses:course_update' pk=course.pk %}">active</a> field?</p>
+  <div class="well">
+    <p>Registed Student(s): {{ population }}</p>
```
no need for the (s).  "Registered Students: 1" makes perfect sense

9921622 replied:
tweaked

tylerecouture commented on src/courses/templates/courses/course_confirm_delete.html:
```diff
@@ -4,14 +4,21 @@
   <input type="submit" value="Delete" class="btn btn-danger" />
 </form>
+{% else %}
+  <p>Unable to delete '{{ course.title }}' as it still has students registered. Perhaps try to disable the <a href="{% url 'courses:course_update' pk=course.pk %}">active</a> field?</p>
```
Change to: `Consider disabling the course by toggling the <a href="{% url 'courses:course_update' pk=course.pk %}">active</a> field instead.`

9921622 replied:
changed

tylerecouture reviewed (changes requested):
A few minor typos, and text update, then should be good!

9921622 commented:
Added requested changes
